core/controllers/base_test.py

```diff
@@ -298,22 +298,28 @@ def test_dev_mode_cannot_be_true_on_production(self):
             Exception, 'DEV_MODE can\'t be true on production.')
         with assert_raises_regexp_context_manager, server_software_swap:
             # This pragma is needed since we are re-importing under
-            # invalid conditions.
+            # invalid conditions. The pylint error messages
+            # 'reimported', 'unused-variable', 'redefined-outer-name' and
+            # 'unused-import' would appear if this line was not disabled.
             import feconf  # pylint: disable-all
 
     def test_valid_pillow_path(self):
+        # We need to re-import appengine_config here to make it look like a
+        # local variable so that we can again re-import appengine_config later.
         import appengine_config
         assert_raises_regexp_context_manager = self.assertRaisesRegexp(
             Exception, 'Invalid path for oppia_tools library: invalid_path')
 
         def mock_os_path_join_for_pillow(*args):
-            """Mocks path for Pillow to test against an invalid path."""
+            """Mocks path for 'Pillow' with an invalid path. This is done by
+            substituting os.path.join with an invalid path. This is needed
```

Comment:
os.path.join is a method, and an "invalid path" is a string, so "substituting os.path.join with an invalid path" does not make sense. Please clarify this sentence.

Reply:
Is it better now?